src/vs/editor/contrib/snippet/test/snippetVariables.test.ts

```diff
@@ -291,6 +292,36 @@ suite('Snippet Variables Resolver', function () {
 		assertVariableResolve3(resolver, 'CURRENT_SECONDS_UNIX');
 	});
 
+	test('Time-based snippet variables resolve to the same values even as time progresses', async function () {
+		const snippetText = `
+			$CURRENT_YEAR
+			$CURRENT_YEAR_SHORT
+			$CURRENT_MONTH
+			$CURRENT_DATE
+			$CURRENT_HOUR
+			$CURRENT_MINUTE
+			$CURRENT_SECOND
+			$CURRENT_DAY_NAME
+			$CURRENT_DAY_NAME_SHORT
+			$CURRENT_MONTH_NAME
+			$CURRENT_MONTH_NAME_SHORT
+			$CURRENT_SECONDS_UNIX
+		`;
+
+		const clock = sinon.useFakeTimers();
+		try {
+			const resolver = new TimeBasedVariableResolver;
+
+			const firstResolve = new SnippetParser().parse(snippetText).resolveVariables(resolver);
+			clock.tick((365 * 24 * 3600 * 1000) + (24 * 3600 * 1000) + (3661 * 1000));  // 1 year + 1 day + 1 hour + 1 minute + 1 second
```

Comment:
Since `_date` is a new constructor argument you could simplify this further, e.g. skip clock-magic and just a fixed date when constructing the resolver

Reply:
I guess it depends on what behaviour you want to test.

The test as it is shows that "the same resolver will return the same values even though time has passed".
Changing the test to just use the constructor would change the test to "the resolver uses the date given".

I agree that the different tests would fail in the same ways, but IMO, the intention is captured better in the current code. 🤷 

Happy to make the change if you want it.

Reply:
> The resolver only has a constructor argument since we don't have "native" private fields.

Not sure what that means? You can define a non-native private field anywhere, not just as ctor-argument-field

Reply:
> I agree that the different tests would fail in the same ways, but IMO, the intention is captured better in the current code. 🤷

Yeah, either way is fine for me. I would have done it different or most likely I wouldn't have written a test ;-)

Reply:
> You can define a non-native private field anywhere, not just as ctor-argument-field

Whoops. Fixed in https://github.com/microsoft/vscode/pull/128857